onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -1146,20 +1120,78 @@ def aten_conv_transpose1d(
     raise NotImplementedError()
 
 
+@torch_op("aten::convolution", trace_only=True)
 def aten_convolution(
-    input: TensorType,
-    weight: TensorType,
-    bias: Optional[TensorType],
+    input: TFloat,
+    weight: TFloat,
+    bias: Optional[TFloat],
     stride: Sequence[int],
-    padding: INT64,
+    padding: Sequence[int],
     dilation: Sequence[int],
     transposed: bool,
-    output_padding: INT64,
+    output_padding: Sequence[int],
     groups: int,
-) -> TensorType:
+) -> TFloat:
     # convolution(Tensor input, Tensor weight, Tensor? bias, int[] stride, SymInt[] padding, int[] dilation, bool transposed, SymInt[] output_padding, int groups) -> Tensor
 
-    raise NotImplementedError()
+    if not isinstance(padding, Sequence):
+        padding = (padding, padding)
+    pads = [*padding, *padding]
+
+    if not isinstance(dilation, Sequence):
+        dilation = (dilation, dilation)
+    dilations = list(dilation)
+
+    if not isinstance(stride, Sequence):
+        stride = (stride, stride)
+    strides = list(stride)
+
+    if bias is None:
+        weight_dim_0 = op.Shape(weight, start=0, end=1)
+        bias_shape = op.Expand(weight_dim_0, op.Constant(value_ints=[1]))
+        zero = op.CastLike(0.0, input)
+        bias = op.Expand(zero, bias_shape)
+
+    result = _aten_convolution_onnx(
+        input, weight, bias, strides=strides, pads=pads, dilations=dilations, transposed=transposed, output_padding=output_padding, groups=groups
+    )
+
+    return result
+
+
+@torch_op("aten::convolution", overload=True)
+def _aten_convolution_onnx(
+    input: TFloat,
+    weight: TFloat,
+    bias: TFloat,
+    strides: Sequence[int],
+    pads: Sequence[int],
+    dilations: Sequence[int],
+    transposed: bool = False,
+    output_padding: Sequence[int] = (0,),
```

Comment:
I don’t think tuple defaults are supported? Since this function is not public, we can make all params required (remove defaults)